### DISCUSSION TITLE
xdrip GATT_CONN_TERMINATE_PEER_USER via recommended settings

### DISCUSSION THREAD

**Discussion by notprogrammer** [Navid200/xDrip#24] [Dexcom]
Hello, i set xdtip setting to recommended from: https://github.com/Navid200/xDrip/wiki/Troubleshoot

But still not working for GATT_CONN_TERMINATE_PEER_USER viz screenshots. Ah, i canot attach screeenshots.

Please help me or send me link where is this solved.

Thank very much.

My Phone: Xiaomi Redmi Note 4 global (Android 7.0 NRD90M with MIUI Global 11.0.2

**Comment by Navid200** [comment]
When you are editing your post here, you will see an edge at the bottom of the message, with "Attach file by ...".  To attach an image, first place your cursor in your message where you want to place the image.  Then, just click on that border (at the bottom) and select the image.   

Please post a screenshot of the system status page.
You can blackout the serial number if you like.  But, please leave the first two characters visible.

**Comment by notprogrammer** [comment]
Sorry, I can see that, I was temporarily blind. It's bad UX for me. :-)

So I'm sending the pictures now. 

Preventive battery replacement did not help either.

Thanks you very much.

![Screenshot_2021-05-31-19-21-39-221_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120225959-86343600-c246-11eb-90a9-ba9dff722ff9.jpg)

![Screenshot_2021-05-31-19-21-33-564_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120225960-86cccc80-c246-11eb-81b0-beacb00ecdd8.jpg)

![Screenshot_2021-05-31-19-21-26-857_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120225962-87fdf980-c246-11eb-827a-e210c2968715.jpg)

![Screenshot_2021-05-31-19-22-04-412_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120225946-816f8200-c246-11eb-8754-164bc54a73ea.jpg)

![Screenshot_2021-05-31-19-22-01-394_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120225950-82a0af00-c246-11eb-8e98-f5f634efc364.jpg)

![Screenshot_2021-05-31-19-21-56-360_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120225954-83d1dc00-c246-11eb-8f53-9c01c86595a5.jpg)

![Screenshot_2021-05-31-19-21-42-166_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120225955-85030900-c246-11eb-9549-b42102ae70be.jpg)

**Reply by Navid200** [reply]
Your phone is either Android 10 or 11, isn't it?

**Reply by notprogrammer** [reply]
No, my phone is Xiaomi Redmi Note 4 Global with Android 6

**Reply by Navid200** [reply]
OK,

Can we also see the Bluetooth settings?

Settings -> Less common settings -> Bluetooth settings

**Reply by notprogrammer** [reply]
Yes, i set all setting to recommended settings from your "troubleshooting" link. In last days i tested on loan transmitter and working perfect. But this transmitter i cannot pair/use or what is the problem. I checked voltage via voltmetter and i get 19,3 V.

**Reply by Navid200** [reply]
19.3V is impossible.   Unless you measured 193mV, or 0.193V.

**Reply by notprogrammer** [reply]
If course. I am not eletrician. :-)

**Reply by Navid200** [reply]
Can you tap on "Forget Device" on the classic status page?
Then, wait 5 minutes.
Then, post the event logs.

**Reply by notprogrammer** [reply]
I'm sorry, I didn't notice you wrote off. Github didn't send me a report. Here are more screenshots.
![Screenshot_2021-05-31-22-59-40-048_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120242049-4c722800-c264-11eb-8009-97da1bcbe5d8.jpg)
![Screenshot_2021-05-31-22-51-56-364_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120242054-4da35500-c264-11eb-8e8e-e787c9d34997.jpg)
![Screenshot_2021-05-31-22-51-40-811_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120242059-4ed48200-c264-11eb-97e9-bf0697632268.jpg)
![Screenshot_2021-05-31-22-51-07-388_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120242063-4f6d1880-c264-11eb-887c-9e811852832e.jpg)

**Reply by Navid200** [reply]
It looks like the battery is dying.  174 days is not bad.

**Reply by notprogrammer** [reply]
Like I said, the batteries are fresh, this is old data.

**Reply by Navid200** [reply]
Are these replaced batteries?  If yes, you may need to re-install them to reduce the resistance.

**Reply by notprogrammer** [reply]
I don't understand how reinstalling the app will help. However, as I say, is the battery information old? Does the status say "last battery queriad 7 days ago"?

**Reply by Navid200** [reply]
I have a hard time understanding you.
Data from 7 days ago says the battery is bad as it shows low voltage B and high resistance.
If the batteries are replaced, and the person who did it did not do a good job, the connection may not be good resulting in high resistance.  If the batteries are redone better improving any issues with the contacts, the resistance may drop increasing voltage B (loaded voltage).

**Reply by notprogrammer** [reply]
Yes, 7 days ago the resistance was high and the voltage weak. Today the batteries were replaced and now the voltage is 19.3 mV

**Reply by Navid200** [reply]
19.3mV is too low.

**Reply by Navid200** [reply]
Use a scanner to see what the signal level is.
https://github.com/Navid200/xDrip/wiki/Bluetooth-Scanner

**Reply by notprogrammer** [reply]
19.3 mV it was actually through a battery replacement, now I have remeasured it and it is 19.45 mV.

My transmitters are:
DexcomBR 19.5 mV (dBm: -63, -56, -49) working perfect
**Dexcom9F 19.45 mV (dBm: -65, -56, -47, -62) we are not working until now, this is what this ticket is about**
DexcomFG 19.5 mV (dBm: -54, -72) non-functional with the same symptoms as the above 9F.

FYI: The BLE scanner was not easy to take a screenshot to make it even bold. Usually I didn't catch it fat for a second. :-)

BTW: The communication of the transmitters therefore takes place non-stop, even if it is not connected to the sensor. When I have three and use only one, isn't it better to remove the batteries from the unused ones so that the transmitter electronics don't wear out unnecessary communication?

![Screenshot_2021-06-02-09-48-39-891_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446089-8a1ca100-c389-11eb-9f5e-91e5f4231a38.jpg)
![Screenshot_2021-06-02-09-48-33-462_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446111-8db02800-c389-11eb-9dc1-16c40d7ca813.jpg)
![Screenshot_2021-06-02-09-45-39-573_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446142-9274dc00-c389-11eb-849e-0601b6dacbd7.jpg)
![Screenshot_2021-06-02-09-44-05-690_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446145-93a60900-c389-11eb-8a12-4b9351d48c5a.jpg)
![Screenshot_2021-06-01-15-19-18-036_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446149-943e9f80-c389-11eb-9c23-8ab2a931677f.jpg)
![Screenshot_2021-06-01-15-15-45-208_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446155-94d73600-c389-11eb-8193-5203417f6954.jpg)
![Screenshot_2021-06-01-15-08-44-559_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446161-96086300-c389-11eb-8eaa-e8ede8449480.jpg)
![Screenshot_2021-06-01-14-53-44-238_com macdom ble blescanner](https://user-images.githubusercontent.com/21248300/120446182-986abd00-c389-11eb-937b-4519b3e16444.jpg)

**Reply by Navid200** [reply]
The Dexcom9F signal level is perfect.  There is a discrepancy between that and the voltages reported on the system status page.

Would you please edit your very first post and and to it the phone brand and model?

**Reply by notprogrammer** [reply]
1) I added information about the phone and the system to the first post.

2) I've been without CGM for a week, I put the functional DexcomBR there and I set the problematic Dexcom 9F to the secondary paralleling of the xdrip application, maybe I don't mind running two applications side by side on one device. In the old application I only had a week old information about the batteries (before replacing them) and I don't have any information about the battery yet.

**Reply by notprogrammer** [reply]
This is a screenshot from new secondary app with default settings.

![Screenshot_2021-06-02-14-28-50-433_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21248300/120480101-f9f15280-c3ae-11eb-9133-b908352c6893.jpg)

**Reply by notprogrammer** [reply]
No changes, stay not working
![Uploading Screenshot_2021-06-02-14-33-47-421_com.eveningoutpost.dexdrip.jpg…]()

**Reply by Navid200** [reply]
Looks like this one isn't working either.
Hoe about the Dexcom app?  
Does any of your transmitters that doesn't work with xDrip work fine with the Dexcom app?

**Reply by notprogrammer** [reply]
Good idea, I'm going to try it.

So I installed an unofficial community version of the Dexcom app.

My status:

- xDrip primary app: DexcomBR working fine
- xDrip secondary app: DexcomFG not working
- Dexcom app: Dexom9F wrote me that "transmitter not found". Does it suggest that I didn't put the transmitter in the sensor, but I have it here on the table? I guess it doesn't matter? Or does it not have to be in the sensor for pairing, it should be paired anyway, it just won't read blood glucose? Should I try to connect it to the sensor and try again like this?

Someone once told me that applications can interfere with each other. Or don't I have to worry?

**Reply by Navid200** [reply]
Ideally, you should not attempt to connect to the same transmitter with xDrip and Dexcom app at the same time.

If you are dealing with a brand new transmitter that has never been used, you may need to place it in a sensor for it to start transmitting.

If DexcomBR is working fine, why not use it then?

**Reply by notprogrammer** [reply]
Not every application tries to connect to a different transmitter. So it's okay.

Not a single transmitter is out of the ordinary, the DexcomBR, which is the only one to work, has replaced the battery several times, but the other two, which have super voltage and signal, are replaced the batteries for the first time.

I can't use DexcomBR because the only one isn't mine and I only borrow it.

We were convinced that the problem is not in the xDrip, maybe it is not in the transmitters either, but it may be in my mobile phone. I will try to install xdrip on my old samsung J3 and try there. You can't help me more. But thank you for your effort. I didn't know you were the creator of xdrip, you did a great thing that helps people who don't have much money like me. That allows me to work on refurbished transmitters.

**Reply by Navid200** [reply]
I'm not the creator of xDrip.
I one of those who tries to help users.

**Reply by notprogrammer** [reply]
aha, I can already see it: forked from NightscoutFoundation / xDrip

The Samsung J3 has the same symptoms as the Xiaomi, so it's not even a cell phone. It's probably the transmitters themselves. Obviously, new batteries, high voltage and a great signal are not enough for proper operation.

I won't bother you anymore, the problem is not in xdrip. You can close it if you don't have a tip for me on what to do with it.

Thank you very much again.

**Reply by Navid200** [reply]
If you look at the system status page with the transmitter that works, very likely, you will see a lower resistance and higher Voltage B.
Whoever replaced the batteries, can improve the connectivity between the battery and the transmitter.  That will lower the resistance and raise Voltage B.  That will solve the problem.
